### ISSUE TITLE
[bug] features.get should not allow fuzzy matches, unless specified

### ISSUE THREAD

**Issue by xgui3783** [FZJ-INM1-BDA/siibra-python#530]
as of `siibra==1.0a4`, feature query via bounding box allow space warping, without explicitly setting any flags. This, IMO, is the incorrect behavior. 

sample code:

```python
import siibra
from siibra.locations.boundingbox import BoundingBox

pts = [[-19.99191860496405,-17.981918604964047,-1.6956886049640476],[19.99191860496405,22.001918604964047,38.28814860496405]]
bbox = BoundingBox(*pts, "mni152")

assert len(siibra.features.get(bbox, "CellbodyStainedSection")) == 0 # fails, 33 != 0
```

this has the comedic aftermath of allow big brain images to be queried in mni space (and so on)

![Screenshot 2024-01-10 164756](https://github.com/FZJ-INM1-BDA/siibra-python/assets/19381783/d38e88f6-dc2a-4812-8bc4-4024dfb97cd3)

**Comment by xgui3783** [comment]
This is a doubly bad for bounding box search, since the warping of bbox is network bound (i.e. takes very long), and for siibra-api, boundingbox queries are the only queries *not* cached

**Comment by AhmetNSimsek** [comment]
This is not a bug as it works as intended after the refactoring. However, we should discuss if we would like to keep this feature or perhaps make it optional in the next dev meeting.

> This is a doubly bad for bounding box search, since the warping of bbox is network bound (i.e. takes very long), and for siibra-api, boundingbox queries are the only queries _not_ cached

Hmm lemme see what can we do.

**Comment by xgui3783** [comment]
I still think this is a downgrade from v0.4. 

in v0.4, it is possible to query all VOI of a specific space. Whilst it was slightly clunky to match the bounding box, it is CPU bound, and relatively fast. 

At the current state, bounding box based query of VOI is borderline unusable. e.g.:

v0.4 (current prod)
```
time curl -o /dev/null 'https://siibra-api-stable.apps.hbp.eu/v3_0/feature/Image?type=CellbodyStainedSection&space_id=minds/core/referencespace/v1.0.0/dafcffc5-4826-4bf1-8ff6-46b8a31ff8e2&bbox=%5B%5B-30.275043349299313,-37.99246234929931,7.9777456507006885%5D,%5B9.709881349299312,1.9924623492993117,47.96267034929931%5D%5D&page=1'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    51  100    51    0     0    134      0 --:--:-- --:--:-- --:--:--   134

real    0m0.384s
user    0m0.042s
sys     0m0.000s
```

v1.x (staging)

```
time curl -o /dev/null 'http://localhost:10081/v3_0/feature/Image?type=CellbodyStainedSection&space_id=minds/core/referencespace/v1.0.0/dafcffc5-4826-4bf1-8ff6-46b8a31ff8e2&bbox=%5B%5B-30.275043349299313,-37.99246234929931,7.9777456507006885%5D,%5B9.709881349299312,1.9924623492993117,47.96267034929931%5D%5D&page=1'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  259k  100  259k    0     0  18172      0  0:00:14  0:00:14 --:--:-- 60480

real    0m14.605s
user    0m0.006s
sys     0m0.000s
```

Keep in mind that this is only one (al beit the worst) example. In prod, siibra-explorer makes a separate query for each of the 8 Image subtypes. Futher, you add concurrent users...

I would request an escape hatch (flag, env var), where VOI query would not wrap bbox of volume.

Otherwise, adopting this version of siibra-python would be difficult for siibra-api

**Comment by xgui3783** [comment]
whilst, #538 should fix this issue, but current main (72367377762ff93a3b66837a3d862c12ba8d7556), the operation is still unusably slow. 

for reference the code I am using is as below:

```python
# bla.py
import sys

FT = "CellbodyStainedSection"

PT1 = (-30.275043349299313, -37.99246234929931, 7.977745650700)
PT2 = (9.709881349299312,1.9924623492993117,47.962670349299)

def main(restrict_space=False):
    import siibra
    from siibra.locations import BoundingBox
    bbox = BoundingBox(PT1, PT2, "mni152")
    print(bbox)
    fts = siibra.features.get(bbox, FT, restrict_space=restrict_space)
    print(len(fts))

if __name__ == "__main__":
    main(*sys.argv[1:])
```

usage:

```sh
$ time python bla.py
[siibra:INFO] Version: 1.0a04
[siibra:WARNING] This is a development release. Use at your own risk.
[siibra:INFO] Please file bugs and issues at https://github.com/FZJ-INM1-BDA/siibra-python.
Loading preconfigured Space instances: 100%|█████████████████████████████████████████████████████| 9/9 [00:00<00:00, 7821.95Space/s]
Bounding box from (-30.28,-37.99,7.98) mm to (9.71,1.99,47.96)mm in MNI 152 ICBM 2009c Nonlinear Asymmetric space
[siibra:INFO] 'CellbodyStainedSection' decoded as feature type/s: ['CellbodyStainedSection'].
Loading preconfigured CellbodyStainedSection instances: 100%|██████████████| 145/145 [00:00<00:00, 24879.28CellbodyStainedSection/s]
Matching CellbodyStainedSection to Bounding box from (-30.28,-37.99,7.98) mm to (9.71,1.99,47.96)mm in MNI 152 ICBM 2009c Nonlinear 
35

real    0m23.419s
user    0m6.616s
sys     0m1.965s
```

restrict space

```
$ time python bla.py 1
[siibra:INFO] Version: 1.0a04
[siibra:WARNING] This is a development release. Use at your own risk.
[siibra:INFO] Please file bugs and issues at https://github.com/FZJ-INM1-BDA/siibra-python.
Loading preconfigured Space instances: 100%|█████████████████████████████████████████████████████| 9/9 [00:00<00:00, 8088.44Space/s]
Bounding box from (-30.28,-37.99,7.98) mm to (9.71,1.99,47.96)mm in MNI 152 ICBM 2009c Nonlinear Asymmetric space
[siibra:INFO] 'CellbodyStainedSection' decoded as feature type/s: ['CellbodyStainedSection'].
Loading preconfigured CellbodyStainedSection instances: 100%|██████████████| 145/145 [00:00<00:00, 26399.88CellbodyStainedSection/s]
Matching CellbodyStainedSection to Bounding box from (-30.28,-37.99,7.98) mm to (9.71,1.99,47.96)mm in MNI 152 ICBM 2009c Nonlinear 
0

real    0m22.676s
user    0m6.560s
sys     0m1.993s

```

**Comment by xgui3783** [comment]
It was discovered that one of the `get_boundingbox` did not use file cache for result.
